line_filtering_node:
  ros__parameters:
    line_sub_topic: "/line/pose_array"
    camera_info_sub_topic: "/cam_down/camera_info"
    dvl_depth_sub_topic: "/dvl/depth"
    pose_array_pub_topic: "/line/pose_array_filtered"
    odom_sub_topic: "/orca/odom"

    debug_visualization: true

    sub_camera_info: "camera_info_sub_"
    sub_depth: "depth_sub_"
    sub_array: "pose_array_sub_"
    pub_topic: "point_cloud_pub_"
    pub_array: "pose_array_pub_"
    pub_point1: "point_1_"
    pub_point2: "point_2_"
    pub_point3: "point_3_"
    pub_point4: "point_4_"

    action_name: "pipeline_following"

    clutter_rate: 0.001
    probability_of_detection: 0.6
    probability_of_survival: 0.98
    gate_threshold: 2.5
    min_gate_threshold: 1.0
    max_gate_threshold: 3.0
    confirmation_threshold: 0.8
    deletion_threshold: 0.2
    initial_existence_probability: 0.25
    std_dynmod: 0.3
    std_sensor: 0.3
    update_interval_ms: 100

    connected_line_threshold: 0.5
    crossing_min_angle: 0.5236 # 30 degrees

    termination_counter_threshold: 30

    line_point_pub_topic: "/line/point"
    line_intersection_pub_topic: "/line/intersection"

    target_frame: "odom"